### PR TITLE
Sync minor improvements for 0.4.0 from the advanced branch

### PR DESCRIPTION
**Pull request by rth**
This aims to add to master 2 PRs from advanced branch that I think should be fairly safe to include in 0.4.0  (#264 )
 -  adding --data-label (#245) 
 - hyperopt fix (#257)

For the other more major PRs in advanced, I think it would be better to include them for 0.5.0 .

Will merge this **without** squashing to preserve history and attributions.

### CODE REVIEW

File: rampwf/utils/testing.py [hunk 1]
```diff
@@ -37,18 +37,31 @@ def assert_title(ramp_kit_dir='.'):
...
 
 
-def assert_data(ramp_kit_dir='.', ramp_data_dir='.'):
+def assert_data(ramp_kit_dir='.', ramp_data_dir='.', data_label=None):
     problem = assert_read_problem(ramp_kit_dir)
-    print_title('Reading train and test files from {}/data ...'.format(
-        ramp_data_dir))
-    X_train, y_train = problem.get_train_data(path=ramp_data_dir)
-    X_test, y_test = problem.get_test_data(path=ramp_data_dir)
+    if data_label is None:
```
Comment (agramfort) [inline rampwf/utils/testing.py:42]: I would be more natural to branch inside get_train_data and get_test_data

I am sure you don't do think to stay backward compatible but maybe can you add a comment to explain this?

File: rampwf/utils/testing.py [hunk 2]
```diff
@@ -172,6 +197,9 @@ def blend_submissions(submissions, ramp_kit_dir='.', ramp_data_dir='.',
         The directory of the ramp-kit to be blended.
     ramp_data_dir : str, default='.'
         The directory of the data.
+    data_label : str, default=None
+        The subdirectory of data in /data and training outputs in
```
Comment (agramfort) [inline rampwf/utils/testing.py:201]:
```suggestion
        The subdirectory of data in {ramp_data_dir}/data and training outputs in
```

File: rampwf/utils/testing.py [hunk 3]
```diff
@@ -172,6 +197,9 @@ def blend_submissions(submissions, ramp_kit_dir='.', ramp_data_dir='.',
...
+        /submissions/<submission>/training_output
```
Comment (agramfort) [inline rampwf/utils/testing.py:202]:
```suggestion
        {ramp_kit_dir}/submissions/<submission>/training_output
```

**Review by agramfort — COMMENTED** [review]
also we should have a changelog / what's new page to document changes

thx @rth

**Comment by rth** [comment]
> also we should have a changelog / what's new page to document changes

Indeed updated the changelog in https://github.com/paris-saclay-cds/ramp-workflow/pull/265, I was thinking to merge that one first to avoid to deal with merge conflicts here.

**Comment by agramfort** [comment]
ok fair enough... merge when happy

>

File: rampwf/utils/testing.py (hunk 1 above)
Comment (rth) [reply rampwf/utils/testing.py]: Indeed, I simplified that this code a bit.

**Comment by albertcthomas** [comment]
Thanks @rth and @agramfort

**Pull request merged**